ragaai_catalyst/tracers/exporters/ragaai_trace_exporter.py

```
@@ -34,26 +34,30 @@ def __init__(self, files_to_zip, project_name, project_id, dataset_name, user_de
+                self.trace_spans[trace_id].append(span_json)
+
+                if span_json["parent_id"] is None:
+                    trace = self.trace_spans[trace_id]
+                    try:
+                        self.process_complete_trace(trace, trace_id)
+                    except Exception as e:
+                        raise Exception(f"Error processing complete trace: {e}")
+                    try:
+                        del self.trace_spans[trace_id]
+                    except Exception as e:
+                        raise Exception(f"Error deleting trace: {e}")
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)